flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/Fabric8FlinkKubeClient.java

```diff
@@ -354,6 +394,29 @@ public KubernetesPod loadPodFromTemplateFile(File file) {
         return new KubernetesPod(this.internalClient.pods().load(file).get());
     }
 
+    public static FlinkKubeClient get(Configuration flinkConfig, Config kubeConfig) {
+        synchronized (LOCK) {
+            final Tuple2<String, String> cacheKey =
+                    Tuple2.of(kubeConfig.getMasterUrl(), kubeConfig.getNamespace());
+            if (CACHE.containsKey(cacheKey)) {
+                CACHE.get(cacheKey).f1++;
+                return CACHE.get(cacheKey).f0;
+            }
+
+            final NamespacedKubernetesClient client = new DefaultKubernetesClient(kubeConfig);
+            final int poolSize =
+                    flinkConfig.get(
+                            KubernetesConfigOptions.KUBERNETES_CLIENT_IO_EXECUTOR_POOL_SIZE);
+            final ExecutorService executorService =
+                    Executors.newFixedThreadPool(
+                            poolSize, new ExecutorThreadFactory("flink-kubeclient-io@" + cacheKey));
+            final Fabric8FlinkKubeClient flinkKubeClient =
+                    new Fabric8FlinkKubeClient(flinkConfig, client, () -> executorService);
+            CACHE.put(cacheKey, Tuple2.of(flinkKubeClient, 1));
+            return flinkKubeClient;
+        }
+    }
```

Comment:
I would first try how it performs without the `CACHE`. I am sceptical whether this is really needed.